### PR TITLE
docs: tweak documentation

### PR DESCRIPTION
**Pull request by richb-hanover**
Several changes to documentation to include more links between pages:
- Add links to keywords for more info;
- Tweak description of strings;
- Add literals to Syntax README;
- Change `sort` to say "tuple"

### CODE REVIEW

File: web/book/src/reference/syntax/README.md
Comment (max-sixty) [inline web/book/src/reference/syntax/README.md:20]: I think this is the main addition in this table? Good idea

File: web/book/src/SUMMARY.md
Comment (max-sixty) [inline web/book/src/SUMMARY.md:13]: I am +1 on making more of a hierarchy here

...but are these literals? I think technically they're not — they are not constant — PRQL changes them...

WDYT?

(we can revert this and merge, or discuss before merging, as you wish)

File: web/book/src/reference/syntax/keywords.md
Comment (max-sixty) [inline web/book/src/reference/syntax/keywords.md:109]: +1

File: web/book/src/reference/syntax/literals.md
Comment (max-sixty) [inline web/book/src/reference/syntax/literals.md]: +1, thanks!

File: web/book/src/reference/syntax/literals.md
Comment (max-sixty) [inline web/book/src/reference/syntax/literals.md]: Great!

File: web/book/src/reference/syntax/literals.md
Comment (max-sixty) [inline web/book/src/reference/syntax/literals.md:107]: Yes better. 

Your grammar is better than mine!

**Review by max-sixty — COMMENTED** [review]
Excellent, pending one question re literals...

File: web/book/src/reference/syntax/README.md
Comment (richb-hanover) [reply web/book/src/reference/syntax/README.md:20]: The major change is inclusion of the remaining items in the Literals page: Numbers, Boolean, Null.

File: web/book/src/SUMMARY.md
Comment (richb-hanover) [reply web/book/src/SUMMARY.md:13]: I agree that s-strings and f-strings aren't strictly literals, but the reader will be comforted to see all the "string-like" things lumped together. (I don't think those string-types would be any easier for newcomers to find farther down in the Syntax table of content.) 

Let's commit as-is and see if anyone comments.

File: web/book/src/reference/syntax/literals.md
Comment (richb-hanover) [reply web/book/src/reference/syntax/literals.md:107]: Probably not, but I have more time to spend on fussy stuff... :-)

File: web/book/src/SUMMARY.md
Comment (max-sixty) [reply web/book/src/SUMMARY.md:13]: I'm hesitant to add something that is wrong... What do others think @PRQL/prql-team ?

Is there a way of adding a bit of hierarchy here while keeping it correct?

File: web/book/src/reference/syntax/README.md
Comment (max-sixty) [reply web/book/src/reference/syntax/README.md:20]: Ah yes, I see, the next couple of lines after `Strings` too.

I'm not really sure `true` / `null` are _syntax_, but don't have a clear view, can see the benefit of having them here, and defer to you.

File: web/book/src/SUMMARY.md
Comment (richb-hanover) [reply web/book/src/SUMMARY.md:13]: In an [earlier life (wayback machine from March 2023)](https://web.archive.org/web/20230313115935/https://prql-lang.org/book/) we had a "Language Features" section with Strings, F-strings and S-strings (sections 5.8, 5.9, 5.10) grouped together. 

Perhaps the Strings paragraphs could be factored from the Literals page into a separate page, with S- and F- string pages (and R-string page?) following, or as a sub-pages? The "Syntax README" would still have a link to the main Strings page, which would continue to refer to the other formats.

Comment (richb-hanover) [reply web/book/src/SUMMARY.md:13]: @max-sixty I don't know if others have responded to this PR/suggestion. What do you think?

Comment (max-sixty) [reply web/book/src/SUMMARY.md:13]: Sorry, was out for a couple of days (2nd child!), just back home :)

> In an [earlier life (wayback machine from March 2023)](https://web.archive.org/web/20230313115935/https://prql-lang.org/book/)

lol

---

I think at the moment two options:
- Change `Literals` to `Literals & Strings`, retain the proposed approach of having F & S strings within thta
- Revert to the existing approach of separating `Literals` from F & S strings

Your call on which (or happy if you want to propose an alternative)

Comment (richb-hanover) [reply web/book/src/SUMMARY.md:13]: OK. I am working on factoring out Strings with sub-headings. Coming soon to a theater near you - probably later today :-)

File: web/book/src/reference/syntax/s-strings.md
Comment (max-sixty) [inline web/book/src/reference/syntax/s-strings.md:95]: whoops!

File: web/book/src/reference/syntax/strings.md
Comment (max-sixty) [inline web/book/src/reference/syntax/strings.md:18]:
```suggestion
use the escape character (`\`), or use more quotes.
```
Is that clearer? I worry that `"\\"` isn't the escape character?

File: web/book/src/reference/syntax/literals.md
Comment (max-sixty) [inline web/book/src/reference/syntax/literals.md:30]: Can we put a link to the Strings page at the bottom of the page (e.g. "See also" or similar), since the normal string is a literal?

**Comment by max-sixty** [comment]
Excellent! A couple of tiny changes and then this is a very nice upgrade

File: web/book/src/reference/syntax/strings.md
Comment (richb-hanover) [reply web/book/src/reference/syntax/strings.md:18]: Nope - the markdown needs to escape the escape character - two `\` are required

File: web/book/src/reference/syntax/literals.md
Comment (richb-hanover) [reply web/book/src/reference/syntax/literals.md:30]: Good point. Done

File: web/book/src/reference/syntax/strings.md
Comment (max-sixty) [reply web/book/src/reference/syntax/strings.md:18]: This works:

`````diff
diff --git a/web/book/src/reference/syntax/strings.md b/web/book/src/reference/syntax/strings.md
index ce103f80..8d0dd919 100644
--- a/web/book/src/reference/syntax/strings.md
+++ b/web/book/src/reference/syntax/strings.md
@@ -15,7 +15,7 @@ # Strings
 **Quoting and escape characters**
 
 To quote a string containing quote characters, use the "other" type of quote, or
-use the escape character ("\\"), or use more quotes.
+use the escape character (`\`), or use more quotes.
 
 ```prql
 from artists
`````

File: web/book/src/reference/syntax/literals.md
Comment (max-sixty) [reply web/book/src/reference/syntax/literals.md:30]: Great, can merge as soon as you push!

**Review by max-sixty — APPROVED** [review]

**Comment by max-sixty** [comment]
Great! Thanks a lot @richb-hanover

**Comment by max-sixty** [comment]
We'll wait for 0.9.4 then

**Pull request merged**
